### PR TITLE
Feature: Moveit Interface Python Core

### PR DESCRIPTION
**Pull request by pinakjani**
This feature includes the implementation of a new core **Moveit2 Python API**.

The Moveit2 Python API is wrapped around by **Interbotix Arm Wrapper** in the `move_arm.py`. 

To use this new Interbotix Arm Moveit Python API one needs to import the `move_arm` module from `interbotix_moveit_interface` package.

### CODE REVIEW

**Comment by pinakjani** [comment]
Fixed known changes. Do we have lint check here?

Looking forward to many changes comments ;)

**Comment by LSinterbotix** [comment]
If you want to pre-emptively run tests, you can build the tests:

```
colcon build --packages-select [packages] --cmake-args -DBUILD_TESTING=ON
```

and run the tests:

```
colcon test --packages-select [packages]
```

and check the logs

```
cat log/latest_test/[package]/stdout_stderr.log
```

**Comment by LSinterbotix** [comment]
This PR should target the `xscobots` branch.

File: interbotix_common_toolbox/interbotix_moveit_interface/interbotix_moveit_interface/move_arm.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+                gripper_pressure_lower_limit=gripper_pressure_lower_limit,
+                gripper_pressure_upper_limit=gripper_pressure_upper_limit,
+            )
+
+    def shutdown(self) -> None:
+        """Destroy the node and shut down all threads and processes."""
+        rclpy.shutdown()
+
+
+class InterbotixArmXSInterface:
```
Comment (LSinterbotix) [inline interbotix_common_toolbox/interbotix_moveit_interface/interbotix_moveit_interface/move_arm.py:125]: Would name this and `InterbotixGripperXSInterface` differently so they don't conflict with the original classes. Maybe something like `InterbotixMoveItArmXSInterface`. Not sure about how verbose that is but

File: interbotix_common_toolbox/interbotix_moveit_interface/interbotix_moveit_interface/move_arm.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+        self.planning_component.set_goal_state(robot_state=robot_state)
+        self.plan_and_execute(sleep_time=3.0)
+        self.logger.info('Reached joint Positions')
+
+    def shutdown(self) -> None:
+        """Destroy the node and shut down all threads and processes."""
+        rclpy.shutdown()
+
+
+class InterbotixGripperXSInterface:
```
Comment (LSinterbotix) [inline interbotix_common_toolbox/interbotix_moveit_interface/interbotix_moveit_interface/move_arm.py:270]: Would suggest making a moveit_gripper.py module and moving this there.

File: interbotix_common_toolbox/interbotix_moveit_interface/interbotix_moveit_interface/move_arm.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+        :param gripper_pressure_lower_limit: (optional) lowest 'effort' that should be applied to
+            the gripper if gripper_pressure is set to 0; it should be high enough to open/close the
+            gripper (~150 PWM or ~400 mA current)
+        :param gripper_pressure_upper_limit: (optional) largest 'effort' that should be applied to
+            the gripper if gripper_pressure is set to 1; it should be low enough that the motor
+            doesn't 'overload' when gripping an object for a few seconds (~350 PWM or ~900 mA)
+        """
+        self.logger = get_logger('moveit_py.gripper')
+
+        # instantiate moveit_py instance and a planning component for the panda_arm
```
Comment (LSinterbotix) [inline interbotix_common_toolbox/interbotix_moveit_interface/interbotix_moveit_interface/move_arm.py:301]: panda arm

File: interbotix_common_toolbox/interbotix_moveit_interface/interbotix_moveit_interface/moveit_interface_core.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    # execute the plan
+    if plan_result:
+        logger.info("Executing plan")
+        robot_trajectory = plan_result.trajectory
+        robot.execute(robot_trajectory, controllers=[])
+    else:
+        logger.error("Planning failed")
+
+    time.sleep(sleep_time)
```
Comment (LSinterbotix) [inline interbotix_common_toolbox/interbotix_moveit_interface/interbotix_moveit_interface/moveit_interface_core.py:48]: should use `get_clock().sleep_for()` instead.

**Review by LSinterbotix — CHANGES_REQUESTED** [review]
We don't want to remove the existing `interbotix_moveit_interface`. Our typical organizational pattern for Python libraries/modules is to create an `interbotix_<something>_modules` package. I would suggest moving all of the moveit_arm and related files into their own package and restoring the state of the `interbotix_moveit_interface` package.

Things like launch files, configuration files, RViz configs, etc. live in whatever main package they're launched from. In this case, I guess they would be located in the interbotix_xscobot_moveit package. Is that an issue? Can we reuse the same configs there with the same functionality?

We also want to allow users to send service requests (torque, get registers, etc.). For this, we typically create an `InterbotixRobotXSCore`. However, this object creates a node. I'm assuming the `moveit.planning.MoveItPy` object does the same. Is there a way we can get around this? Can we refactor all the older code so a node is only created if we don't already have one/one isn't passed to the object? Can we also only have one MoveItPy instance per process? What do you think?

**Pull request closed without merging**